components/script/dom/macros.rs

```diff
@@ -632,3 +632,33 @@ macro_rules! handle_potential_webgl_error {
         handle_potential_webgl_error!($context, $call, ());
     };
 }
+
+macro_rules! impl_rare_data (
+    ($type:ty) => (
+        fn init_rare_data(&self) {
```

Comment:
This should just be inlined in `rare_data_mut`, and `rare_data_mut` should probably be called `ensure_rare_data` (to highlight the lazy initialization) and should not return an Option.